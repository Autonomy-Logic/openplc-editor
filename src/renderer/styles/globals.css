@tailwind base;
@tailwind components;
@tailwind utilities;

::-webkit-scrollbar {
  display: flex;
  height: 0.25rem;
  width: 0.25rem;
  border-radius: 0.5rem;
  background-color: rgb(221 226 232 / var(--tw-bg-opacity));
  transition-property: color, background-color, border-color, text-decoration-color, fill, stroke;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 160ms;
  @apply dark:bg-neutral-850;
}

::-webkit-scrollbar-thumb {
  border-radius: 0.5rem;
  background-color: var(--primary-medium);
  transition-property: color, background-color, border-color, text-decoration-color, fill, stroke;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 700ms;
  @apply bg-opacity-30 hover:bg-brand-medium hover:bg-opacity-100;
}

::-webkit-scrollbar-corner {
  background-color: transparent;
}

@layer base {
  :root {
    --primary-default: #0464fb;
    --primary-light: #b4d0fe;
    --primary-medium: #0350c9;
    --primary-medium-dark: #023c97;
    --primary-dark: #011e4b;
    /* Fallback variables to set the height of the titlebar based on OS */
    --oplc-title-bar-height: 2rem;
    --fallback-darwin-title-bar-height: 1.75rem;
  }

  /* This can be useful to allocate the main container */

  #folder-root p:hover ~ svg #folder-path {
    fill: var(--primary-medium-dark);
  }

  button {
    outline: 2px solid transparent;
    outline-offset: 2px;
  }
}

@layer components {
  .box {
    @apply border border-brand-light shadow-oplc dark:border-brand-medium-dark dark:shadow-oplc-dark;
  }
  /* Classes for the custom titlebar*/
  .oplc-titlebar-container {
    position: absolute;
    top: 0;
    width: 100%;
    height: var(--oplc-title-bar-height);
  }

  .oplc-titlebar-content {
    position: absolute;
    top: 0;
    display: flex;
    user-select: none;
    height: 100%;
    background-color: var(--primary-dark);
    left: 0;
    width: 100%;
  }

  .oplc-titlebar-drag-region {
    app-region: drag;
  }

  .oplc-main-content {
    position: absolute;
    display: flex;
    left: 0;
    right: 0;
    bottom: 0;
    top: var(--oplc-title-bar-height);
    overflow: hidden;
  }
  input[type='number']::-webkit-outer-spin-button,
  input[type='number']::-webkit-inner-spin-button {
    appearance: none;
    margin: 0;
  }
  input[type='number'] {
    appearance: textfield;
  }
}

@layer utilities {
  .press-animated {
    @apply border-[none] transition-[0.2s] duration-[all] active:scale-[0.90];
  }

  .no-drag-window {
    -webkit-app-region: no-drag;
  }
}

.apexcharts-yaxis-label tspan,
.apexcharts-xaxis-label tspan {
  @apply fill-neutral-1000 !text-[10px] dark:fill-neutral-50;
}
