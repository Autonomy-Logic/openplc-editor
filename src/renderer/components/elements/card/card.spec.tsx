import { describe, expect, it } from '@jest/globals'
import { render } from '@testing-library/react'
import Card from './index'

// Test suite for the Root Component
describe('Root Component', () => {
	// Test to ensure the Root component renders correctly
	test('renders correctly', () => {
		// Render the Card.Root component and get the container
		const { container } = render(<Card.Root />)
		// Compare the rendered container with a snapshot
		expect(container).toMatchSnapshot()
	})

	// Test to check if HTMLAttributes are passed to the Root component
	test('passes HTMLAttributes to the component', () => {
		// Define a custom prop value
		const customProp = 'custom-prop-value'
		// Render the Card.Root component with specific HTMLAttributes
		const { getByTestId } = render(
			<Card.Root data-testid='root-test-id' data-custom-prop={customProp} />
		)

		// Get the Root element by test ID
		const rootElement = getByTestId('root-test-id')
		// Retrieve the custom prop value from the element's attributes
		const rootProps = rootElement.getAttribute('data-custom-prop')

		// Ensure that the Root element exists and has the correct custom prop value
		expect(rootElement).toBeTruthy()
		expect(rootProps).toBe(customProp)
	})
})

// Test suite for the Label Component
describe('Label Component', () => {
	// Test to ensure the Label component renders correctly
	test('renders correctly', () => {
		// Render the Card.Label component with specific title and description
		const { container } = render(
			<Card.Label title='Test title' description='Test description' />
		)
		// Compare the rendered container with a snapshot
		expect(container).toMatchSnapshot()
	})

	// Test to check if HTMLAttributes are passed to the Label component
	test('passes HTMLAttributes to the component', () => {
		// Define a custom prop value
		const customProp = 'custom-prop-value'
		// Render the Card.Label component with specific HTMLAttributes
		const { getByTestId } = render(
			<Card.Label
				data-testid='label-test-id'
				data-custom-prop={customProp}
				title='Test title'
				description='Test description'
			/>
		)

		// Get the Label element by test ID
		const labelElement = getByTestId('label-test-id')
		// Retrieve the custom prop value from the element's attributes
		const labelProps = labelElement.getAttribute('data-custom-prop')

		// Ensure that the Label element exists and has the correct custom prop value
		expect(labelElement).toBeTruthy()
		expect(labelProps).toBe(customProp)
	})
})

// Test suite for the Preview Component
describe('Preview Component', () => {
	// Test to ensure the Preview component renders correctly
	test('renders correctly', () => {
		// Render the Card.Preview component with a specific source
		const { container } = render(<Card.Preview source='Test source' />)
		// Compare the rendered container with a snapshot
		expect(container).toMatchSnapshot()
	})

	// Test to check if HTMLAttributes are passed to the Preview component
	test('passes HTMLAttributes to the component', () => {
		// Define a custom prop value
		const customProp = 'custom-prop-value'
		// Render the Card.Preview component with specific HTMLAttributes
		const { getByTestId } = render(
			<Card.Preview
				data-testid='preview-test-id'
				data-custom-prop={customProp}
				source='Test source'
			/>
		)

		// Get the Preview element by test ID
		const previewElement = getByTestId('preview-test-id')
		// Retrieve the custom prop value from the element's attributes
		const previewProps = previewElement.getAttribute('data-custom-prop')

		// Ensure that the Preview element exists and has the correct custom prop value
		expect(previewElement).toBeTruthy()
		expect(previewProps).toBe(customProp)
	})
})
