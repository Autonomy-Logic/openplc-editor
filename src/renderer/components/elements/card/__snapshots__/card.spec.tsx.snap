// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component renders correctly 1`] = `
<div>
  <div
    class="flex flex-col items-start justify-start gap-[2px] cursor-pointer"
  >
    <h3
      class="relative leading-4"
    >
      Test title
    </h3>
    <p
      class="relative text-[10px] leading-3 opacity-40"
    >
      Test description
    </p>
  </div>
</div>
`;

exports[`Preview Component renders correctly 1`] = `
<div>
  <img
    alt="card-preview"
    class="self-stretch flex-1 relative rounded-md max-w-full overflow-hidden max-h-full object-cover cursor-pointer"
    src="Test source"
  />
</div>
`;

exports[`Root Component renders correctly 1`] = `
<div>
  <div
    class="mr-7 xl:mr-[15px] xxl:mr-[30px] relative flex flex-1 flex-col items-start justify-start bg-brand-medium w-full min-w-[224px] h-[180px] rounded-lg px-[6px] pt-[6px] pb-[10px] gap-[10px] text-left text-xs text-white font-caption hover:bg-brand-medium-dark cursor-pointer"
  />
</div>
`;
